pkg/skaffold/event/event.go

```diff
@@ -310,7 +310,7 @@ func BuildCanceled(imageName string) {
 func BuildFailed(imageName string, err error) {
 	aiErr := sErrors.ActionableErr(sErrors.Build, err)
 	handler.stateLock.Lock()
-	handler.state.BuildState.StatusCode = aiErr.ErrCode
+	handler.state.BuildState.StatusCode = ignoreCancelled(handler.state.BuildState.StatusCode, aiErr.ErrCode)
```

Comment:
`ignoreCancelled(handler.state.BuildState.StatusCode, aiErr.ErrCode)`

This makes me think that we're updating the `BuildState.StatusCode` in the wrong place since it seems to denote the status of the build sequence of several artifacts as a whole whereas `BuildFailed` is called for every artifact build failure.

Can we split out:
```go
// BuildFailed notifies that a build has failed.
func BuildFailed(imageName string, err error) {
	aiErr := sErrors.ActionableErr(sErrors.Build, err)
	handler.handleBuildEvent(&proto.BuildEvent{
		Artifact:      imageName,
		Status:        Failed,
		Err:           err.Error(),
		ErrCode:       aiErr.ErrCode,
		ActionableErr: aiErr})
}

// BuildSequenceFailed notifies that the build sequence has failed.
func BuildSequenceFailed(err error) {
	aiErr := sErrors.ActionableErr(sErrors.Build, err)
	handler.stateLock.Lock()
	handler.state.BuildState.StatusCode = aiErr.ErrCode
	handler.stateLock.Unlock()
}
```

and call `BuildSequenceFailed` in `scheduler.go` line 70
```go
if err := g.Wait(); err != nil {
	event.BuildSequenceFailed(err)
	return nil, err
}
```

Reply:
Yes. i think with the error group approach it would work.